src/actors/objects/mod.rs

```diff
@@ -265,13 +293,15 @@ impl CacheItemRequest for FetchFileDataRequest {
                             .filter_map(Result::ok)
                             .find(|object| object_id.match_object(object));
 
-                        // If we do not find the desired object in this archive - either
-                        // because we can't parse any of the objects within, or because none
-                        // of the objects match the identifier we're looking for - we return
-                        // early.
                         let object = match object_opt {
                             Some(object) => object,
-                            None => return Ok(CacheStatus::Negative),
+                            None => {
+                                if archive.objects().all(|r| r.is_err()) {
+                                    return Ok(CacheStatus::Malformed);
+                                } else {
+                                    return Ok(CacheStatus::Negative);
+                                }
+                            }
```

Comment:
This fixes a bug where if an archive contained only malformed objects it would show up as missing rather than malformed.  Arguably this isn't aggressive enough and should return malformed if the object was not found and **any** member of the archive failed to be parsed.